base/exporter.go

```diff
@@ -0,0 +1,135 @@
+package base
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type SgwStats struct {
+	GlobalStats GlobalStat `json:"global_stats"`
+	DbStats     map[string]DbStats
+}
+
+func (s SgwStats) String() string {
+	bytes, _ := JSONMarshal(s)
+	return string(bytes)
+}
+
+type GlobalStat struct {
+	ResourceUtilization ResourceUtilization `json:"resource_utilization"`
+}
+
+type ResourceUtilization struct {
+	AdminNetworkInterfaceBytesReceived  *sgwIntStat `json:"admin_network_interface_bytes_recv"`
+	AdminNetworkInterfaceBytesSent      *sgwIntStat `json:"admin_network_interface_bytes_sent"`
+	ErrorCount                          *sgwIntStat `json:"error_count"`
+	GoMemstatsHeapAlloc                 *sgwIntStat `json:"go_memstats_heapalloc"`
+	GoMemstatsHeapIdle                  *sgwIntStat `json:"go_memstats_heapidle"`
+	GoMemstatsHeapInUse                 *sgwIntStat `json:"go_memstats_heapinuse"`
+	GoMemstatsHeapReleased              *sgwIntStat `json:"go_memstats_heapreleased"`
+	GoMemstatsPauseTotalNS              *sgwIntStat `json:"go_memstats_pausetotalns"`
+	GoMemstatsStackInUse                *sgwIntStat `json:"go_memstats_stackinuse"`
+	GoMemstatsStackSys                  *sgwIntStat `json:"go_memstats_stacksys"`
+	GoMemstatsSys                       *sgwIntStat `json:"go_memstats_sys"`
+	GoroutinesHighWatermark             *sgwIntStat `json:"goroutines_high_watermark"`
+	NumGoroutines                       *sgwIntStat `json:"num_goroutines"`
+	ProcessMemoryResident               *sgwIntStat `json:"process_memory_resident"`
+	PublicNetworkInterfaceBytesReceived *sgwIntStat `json:"pub_net_bytes_recv"`
+	PublicNetworkInterfaceBytesSent     *sgwIntStat `json:"pub_net_bytes_sent"`
+	SystemMemoryTotal                   *sgwIntStat `json:"system_memory_total"`
+	WarnCount                           *sgwIntStat `json:"warn_count"`
+	CpuPercentUtil                      *sgwIntStat `json:"cpu_percent_util"`
+}
+
+type DbStats struct {
+}
+
+type sgwIntStat struct {
+	statFQDN      string
+	statDesc      *prometheus.Desc
+	dbName        string
```

Comment:
As discussed, since we also need to support labels like "replication":"replicationName", probably need to change this to a generic 'labels' map.

Reply:
I think this still applies.  It's reasonable to have the constructor just accept dbName for ease of use, but internally it seems more appropriate to store slices of label names/label values.